src/App.tsx

```diff
@@ -1,22 +1,143 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useMemo, useRef, useState,
+} from 'react';
+import { getTodos, addTodo, deleteTodo } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
 
+import { TodoList } from './components/TodoList';
+import { NewTodoForm } from './components/NewTodoField';
+import { Footer } from './components/Footer';
+import { Error } from './components/Error';
+import { Todo } from './types/Todo';
+import { Filters } from './types/Filters';
+import { Errors } from './types/Errors';
+
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string>('');
+  const [filter, setFilter] = useState<string>(Filters.All);
+  const [isAdding, setIsAdding] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingTodoId, setDeletingTodoId] = useState(0);
+  const [isClearCompleted, setClearCompleted] = useState(false);
+
+  function handleError(errorType: string) {
+    setError(errorType);
+    setTimeout(() => {
+      setError('');
+    }, 3000);
+  }
 
   useEffect(() => {
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      getTodos(user.id)
+        .then((todosFromServer) => {
+          setTodos(todosFromServer);
+        })
```

Comment:
```suggestion
        .then(setTodos)
```
just FYI it may look like this